src/services/services.ts

```diff
@@ -5288,6 +5294,149 @@ namespace ts {
             return getDefinitionFromSymbol(type.symbol, node);
         }
 
+         /// Goto implementation
+        function getImplementationAtPosition(fileName: string, position: number): ImplementationLocation[] {
+            synchronizeHostData();
+
+            const entries: ImplementationLocation[] = [];
+            const node = getTouchingPropertyName(getValidSourceFile(fileName), position);
+            const typeChecker = program.getTypeChecker();
+
+            if (node.parent.kind === SyntaxKind.ShorthandPropertyAssignment) {
+                const entry = getReferenceEntryForShorthandPropertyAssignment(node, typeChecker);
+                entries.push({
+                    textSpan: entry.textSpan,
+                    fileName: entry.fileName
+                });
+            }
+            else if (definitionIsImplementation(node, typeChecker)) {

```

Comment:
Yep, I believe that case should fall into the "definition is implementation" camp. Pretty much everything except for things that can have multiple implementations (e.g. interfaces, classes, unions/intersections plus members of all those) also just call "Go to Definition".